memo_app.rb

```diff
@@ -60,28 +85,26 @@ def save_memos(memos)
     erb :edit
   else
     status 404
-    erb :error_404
+    erb :error_not_found
   end
 end
 
 patch '/memos/:id' do
-  memos = load_memos
-  id = params[:id]
-  @memo = memos[id]
+  @id = params[:id]
+  @memo = load_memos[@id]
   if @memo
-    memos[id]['title'] = params[:title]
-    memos[id]['content'] = params[:content]
-    save_memos(memos)
+    title = params[:title]
+    content = params[:content]
+    update_memo(title, content, @id)
     redirect '/memos'
   else
     status 404
-    erb :error_404
+    erb :error_not_found
   end
 end
 
 delete '/memos/:id' do
-  memos = load_memos
-  memos.delete(params[:id])
-  save_memos(memos)
+  @id = params[:id]
+  delete_memo(@id)
```

Comment:
前回指摘が漏れていたかもですが、

```suggestion
  delete_memo(params[:id])
```

でよさそうに思いました。